AllASPFuncs.py

```diff
@@ -1343,4 +1341,40 @@ def contourGrid(image, numlevels = 5, subsize = 4):
         x_totalgrid.extend(xg)
         y_totalgrid.extend(yg)
 
-    return y_totalgrid, x_totalgrid
\ No newline at end of file
+    return y_totalgrid, x_totalgrid
+
+def saveLightcurves(ID, exposures, sn_path, confusion_metric, use_real_images, detim, supernova, X, use_roman, band):   
```

Comment:
I suggest `save_lightcurve` for the function name.  You're using '_' style for most other things.  Switching to camelCase here is an odd deviation.

Reply:
Consider breaking this up into two functions
`make_lightcurve`
`save_lightcurve`

Reply:
Add an argument for the path to write results to.  Perhaps it defaults to `./results`?